src/framework/ui/view/keynavigationevent.h

```diff
@@ -0,0 +1,62 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_UI_KEYNAVIGATIONEVENT_H
+#define MU_UI_KEYNAVIGATIONEVENT_H
+
+#include <QObject>
+#include "ui/ikeynavigation.h"
+
+namespace mu::ui {
+class KeyNavigationEvent : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(Type type READ type CONSTANT)
+    Q_PROPERTY(bool accepted READ accepted WRITE setAccepted NOTIFY acceptedChanged)
+
+public:
+    explicit KeyNavigationEvent(QObject* parent = nullptr);
+
+    //! NOTE Please sync with IKeyNavigation::Event::Type
+    enum Type {
+        Undefined = 0,
+        Left,
+        Right,
+        Up,
+        Down,
+        Escape
+    };
+    Q_ENUM(Type)
+
+    void setEvent(IKeyNavigation::EventPtr event);
+
+    Type type() const;
+    bool accepted() const;
+
+public slots:
+    void setAccepted(bool accepted);
+
+signals:
+    void acceptedChanged(bool accepted);
+
+private:
+    IKeyNavigation::EventPtr m_event;
```

Comment:
Not sure
Raw ptr needs initialise
```
IKeyNavigation::Event* m_event = nullptr;
```
SomeThingPtr (without *) - for us this is always a smart pointer (usually shared_ptr), does not require initialization, because it is self-initializing.

I don't mind initializing smart pointers for consistency, but then it should be done everywhere. This is not the case now, in most places smart pointers are not initialized explicitly